pandas/tests/tools/test_to_datetime.py

```diff
@@ -2530,3 +2530,14 @@ def test_empty_string_datetime_coerce__unit():
     # verify that no exception is raised even when errors='raise' is set
     result = to_datetime([1, ""], unit="s", errors="raise")
     tm.assert_index_equal(expected, result)
+
+
+@pytest.mark.parametrize("cache", [True, False])
+def test_to_datetime_monotonic_increasing_index(cache):
+    # GH28238
+    times = pd.date_range(pd.Timestamp("1980"), periods=50, freq="YS")
+    times = times.to_frame(index=False, name="DT").sample(50)
```

Comment:
Could you set a `random_state` in sample on the very slim chance this doesn't shuffle the index?

Reply:
Should work now